src/main/java/dal/fatih/todoly/Main.java

```diff
@@ -46,67 +45,54 @@ public static Task handleCreateTask() {
         }
         return handleCreateTask();
     }
-
     public static void listAllTasks() {
         if (tasks.isEmpty()) {
             System.out.println("Task list is empty");
         } else {
-            for (Task task : tasks) {
-                System.out.println("ID: " + task.getId() + " TITLE: " + task.getTitle() +
-                        " DUE DATE: " + dateFormatter(task.getDate()));
+            for (Map.Entry task : tasks.entrySet()) {
+                System.out.println("ID: " + task.getKey() + task.getValue());
             }
         }
     }
-
     public static void showTaskDetails() {
-        if (tasks.isEmpty()) {
-            System.out.println("Task list is empty");
-        } else {
-            System.out.println("Task Id :");
-            String taskId = scn.nextLine();
-            Task task = null;
-            for (Task task_ : tasks) {
-                if (task_.getId().equals(taskId)) {
-                    task = task_;
-                }
-            }
-            if (task != null) {
-                System.out.println("ID: " + task.getId() + "\nTITLE: " + task.getTitle() +
-                        "\nDESCRIPTION: " + task.getDescription() +
-                        "\nDUE DATE: " + dateFormatter(task.getDate()));
-            } else {
-                System.out.println("Task not found");
+        System.out.print("Task Id :");
+        String taskId = scn.nextLine();
+        String task = null;
+        for (Map.Entry task_ : tasks.entrySet()) {
+            if (task_.getKey().toString().equals(taskId)) {
+                task = "ID= " + task_.getKey() + task_.getValue().toString();
             }
         }

```

Comment:
You don't use the map for the only reason why you should use it. This code is working exactly as it was with list. No difference. Try to figure out why and how you should use map